vsock-proxy/parent/src/parent.rs

```diff
@@ -186,30 +187,51 @@ async fn send_enclave_exit(enclave_port: &mut AsyncVsockStream) -> Result<(), St
 }
 
 fn filter_env_variables(orig_env_path: PathBuf) -> Result<Vec<(String, String)>, String> {
-    let mut runtime_vars: Vec<(String, String)> = env::vars().collect();
-
+    let mut runtime_vars: HashMap<String, String> = env::vars().collect();
     info!("Found the following environment variables in parent : {:?}", runtime_vars);
 
-    // Filter the environment variables sent from the parent. The ORIG_ENV_LIST_PATH file contains
-    // the list of variables which were set in the parent at conversion time. Rules for filtering:
-    //  - Always pass HOSTNAME
-    //  - Keep all environment variables which are new (i.e. not present in the parent at conversion
-    //    time)
-    //  - Keep environment variables which were present in the parent container but their values
-    //    have now been updated. The exception to this rule is for the PATH variable.
+    // "_" is a special var setup by bash that points to the currently executed binary
+    // to not leak the binary path into the enclave we remove this var explicitly.
+    runtime_vars.remove("_");
```

Comment:
The bash man page states:
```
       _      At shell startup, set to the absolute pathname used to invoke the shell or shell script being executed as passed in the environment or argument list.
```
This seems to imply that there always should be a `_` environment variable. Is there a test in our CI to verify that this still works correctly?